hs_file_types/models/generic.py

```diff
@@ -0,0 +1,159 @@
+                                   style="display: none;",
+                                   onclick="metadata_update_ajax_submit("
+                                           "'id-coverage-spatial-filetype');")
+
+        template = Template(root_div.render())
+        context_dict = dict()
+        temp_cov_form = self.get_temporal_coverage_form()
+        spatial_cov_form = self.get_spatial_coverage_form()
+        update_action = "/hsapi/_internal/GenericLogicalFile/{0}/{1}/{2}/update-file-metadata/"
+        create_action = "/hsapi/_internal/GenericLogicalFile/{0}/{1}/add-file-metadata/"
+
+        element_name = "coverage"
+        if self.temporal_coverage or self.spatial_coverage:
+            if self.temporal_coverage:
+                temp_action = update_action.format(self.logical_file.id, element_name,
+                                                   self.temporal_coverage.id)
+                temp_cov_form.action = temp_action
+            else:
+                temp_action = create_action.format(self.logical_file.id, element_name)
+                temp_cov_form.action = temp_action
+
+            if self.spatial_coverage:
+                spatial_action = update_action.format(self.logical_file.id, element_name,
+                                                      self.spatial_coverage.id)
+                spatial_cov_form.action = spatial_action
+            else:
+                spatial_action = create_action.format(self.logical_file.id, element_name)
+                spatial_cov_form.action = spatial_action
+        else:
+            action = create_action.format(self.logical_file.id, element_name)
+            temp_cov_form.action = action
+            spatial_cov_form.action = action
+
+        context_dict["temp_form"] = temp_cov_form
+        context_dict["spatial_form"] = spatial_cov_form
+        context = Context(context_dict)
+        rendered_html = template.render(context)
+        # file level form field ids need to changed so that they are different from
+        # the ids used at the resource level for the same type of metadata elements
+        rendered_html = rendered_html.replace("div_id_start", "div_id_start_filetype")
+        rendered_html = rendered_html.replace("div_id_end", "div_id_end_filetype")
+        rendered_html = rendered_html.replace("id_start", "id_start_filetype")
+        rendered_html = rendered_html.replace("id_end", "id_end_filetype")
+
+        for spatial_element_id in ('div_id_northlimit', 'div_id_southlimit', 'div_id_westlimit',
+                                   'div_id_eastlimit'):
+            rendered_html = rendered_html.replace(spatial_element_id,
+                                                  spatial_element_id + "_filetype", 1)
+        for spatial_element_id in ('div_id_type', 'div_id_north', 'div_id_east'):
+            rendered_html = rendered_html.replace(spatial_element_id,
+                                                  spatial_element_id + "_filetype", 1)
+
```

Comment:
These string replacements need be done in this particular order. Otherwise, same element gets replaced more than once.  I remember having one loop to do the replacements and noticed this problem.